distributed/worker.py

```diff
@@ -1494,9 +1494,12 @@ async def start(self):
 
         setproctitle("dask-worker [%s]" % self.address)
 
-        await asyncio.gather(
+        plugins_msgs = await asyncio.gather(
             *(self.plugin_add(plugin=plugin) for plugin in self._pending_plugins)
         )
```

Comment:
It's also possible for `plugin_add` to raise other exceptions (e.g. unpickling errors). What would happen in these cases? I assume it is failing hard but I would like to confirm. Do we have tests for this?

Reply:
Added test for that in https://github.com/dask/distributed/pull/4298/commits/d3239133d1b0adc7c4732b5b60fd8fb60231927c .